### ISSUE TITLE
Sprites from TextureAtlas get squished when rendered

### ISSUE THREAD

**Issue by Wulf0x67E7** [bevyengine/bevy#737]
bevy = "0.2.1"
Windows 10 v2004

I was playing with the sprite_sheet.rs example code. After updating it slightly to work with the current API, I noticed that the sprites were getting squished in the x-axis by a factor of seven - which also is the number of sprites in the spritesheet.

I have looked at the TextureAtlas::from_grid source and it seems to correctly set the sprites sizes to the passed Vec2. The bug seems to be somewhere in the renderer.

The code with my fixes:

```rust
use bevy::prelude::*;

fn main() {
    App::build()
        .add_default_plugins()
        .add_startup_system(setup.system())
        .add_system(animate_sprite_system.system())
        .run();
}

fn animate_sprite_system(
    texture_atlases: Res<Assets<TextureAtlas>>,
    mut query: Query<(&mut Timer, &mut TextureAtlasSprite, &Handle<TextureAtlas>)>,
) {
    for (timer, mut sprite, texture_atlas_handle) in &mut query.iter() {
        if timer.finished {
            let texture_atlas = texture_atlases.get(texture_atlas_handle).unwrap();
            sprite.index = ((sprite.index as usize + 1) % texture_atlas.textures.len()) as u32;
        }
    }
}

fn setup(
    mut commands: Commands,
    asset_server: Res<AssetServer>,
    mut texture_atlases: ResMut<Assets<TextureAtlas>>,
) {
    let texture_handle = asset_server
        .load("res/textures/gabe-idle-run.png") // now looks in root directly instead of assets subdirectory
        .expect("couldn't load file"); // now gives error on fail - ^ but not on "file not found"??
    let texture_atlas = TextureAtlas::from_grid(texture_handle, Vec2::new(24.0, 24.0), 7, 1);
    let texture_atlas_handle = texture_atlases.add(texture_atlas);
    commands
        .spawn(Camera2dComponents::default())
        .spawn(SpriteSheetComponents {
            texture_atlas: texture_atlas_handle,
            // Vec3::new(6.0, 6.0, 6.0) results in squishing, Vec3::new(6.0 * 7.0, 6.0, 6.0) renders as Vec3::new(6.0, 6.0, 6.0) should
            transform: Transform::from_non_uniform_scale(Vec3::new(6.0, 6.0, 6.0)), // from_scale for f32 and Vec3 now separate  functions
            ..Default::default()
        })
        .with(Timer::from_seconds(0.1, true));
}
```
Squished:
![image](https://user-images.githubusercontent.com/35270937/97279034-bf34a400-183a-11eb-8bf2-15584d5a9206.png)

Should be:
![image](https://user-images.githubusercontent.com/35270937/97279120-dbd0dc00-183a-11eb-9a27-8c2c4d6b967c.png)

**Comment by Omnisthetics** [comment]
In `0.2.1` the size parameter of `TextureAtlas::from_grid` represents the size of the entire texture rather than the size of tiles/individual sprites. 

Fetching the latest `master` should fix it or, if you want to stick with `0.2.1`, setting the size to `Vec2::new(168.0, 24.0)` should also fix it.

**Comment by Wulf0x67E7** [comment]
Thanks for the answer, both solutions worked perfectly.
Seems the size parameter also doesn't specify the pixel-size of the source, but rather the desired size for rendering. Setting it to Vec2::new(7.0, 1.0) in "0.2.1" / Vec2::new(1.0, 1.0) in master also worked and lets you set the precise rendering pixel-size in the character transform.
